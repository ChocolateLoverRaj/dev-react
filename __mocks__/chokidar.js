// Mocker for chokidar
import EventEmitter from 'eventemitter3'
import sinon from 'sinon'

const chokidar = {}

export class FSWatcher extends EventEmitter { };

FSWatcher.prototype.close = sinon.fake(() => Promise.resolve())

chokidar.watch = sinon.fake(() => new FSWatcher())

export default chokidar
