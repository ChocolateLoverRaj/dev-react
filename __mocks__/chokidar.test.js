/* eslint-env jest */

import chokidar, { FSWatcher, _reset, _failClose } from './chokidar.js'

afterEach(_reset)

test('chokidar.watch', () => {
  const args = ['a', 'b']
  chokidar.watch(...args)
  expect(chokidar.watch.calledOnceWith(...args)).toBe(true)
})

test('FSWatcher.prototype.close', async () => {
  const watcher = chokidar.watch()
  await watcher.close()
  expect(FSWatcher.prototype.close.calledOnce).toBe(true)
})

test('_failClose', async () => {
  _failClose()
  const watcher = chokidar.watch()
  await expect(watcher.close()).rejects.toMatchSnapshot()
})

test('_reset', async () => {
  await chokidar.watch().close()
  _failClose()
  _reset()
  expect(FSWatcher.prototype.close.notCalled).toBe(true)
  expect(chokidar.watch.notCalled).toBe(true)
  await chokidar.watch().close()
})
