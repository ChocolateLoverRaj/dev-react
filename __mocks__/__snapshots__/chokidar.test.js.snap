// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_failClose 1`] = `[Error]`;
